### ISSUE TITLE
Bundles mappings in fuse@v4

### ISSUE THREAD

**Issue by avizcaino** [fuse-box/fuse-box#1957]
Hi! I've been struggling a lot with custom bundles configuration in fuse@v4 and couldn't find the solution. I've prepared a small repo with my configuration in a monorepo environment and using typescript and lit-element.

Repo: https://github.com/avizcaino/fuse-bundler

Here is my fuse config:

```typescript
export const fuse = (
  entry: string,
  baseStyles: string,
  webIndex: string,
  devServer: boolean,
  publicPath = "/",
  env: any = {}
) =>
  fusebox({
    cache: false,
    compilerOptions: {
      tsConfig: "tsconfig.json",
    },
    devServer,
    hmr: true,
    entry,
    env: { ...process.env, ...env },
    target: "browser",
    watcher: {
      root: [workspaceRoot],
    },
    stylesheet: {
      macros: {
        "~": join(__dirname, "../../../"),
      },
      autoImport: [{ file: baseStyles, capture: "packages/*/src" }],
    },
    plugins: [
      pluginSass("*.scss", { asText: true }),
      pluginLink(/.+\.png/, { useDefault: true }),
      pluginLink(/.+\.svg/, { useDefault: true }),
    ],
    webIndex: { template: webIndex, publicPath },
    resources: {
      resourcePublicRoot: `${publicPath}resources/`,
      resourceFolder: `resources/`,
    },
  });

fuse('demo/index.ts', '../styles/styles.scss', 'demo/index.html', false).runProd({
  uglify: true,
  target: 'browser',
  bundles: {
    app: './app.$hash.js',
    mapping: [
      { matching: 'webcomponents*', target: './vendor.webcomponents.$hash.js' },
      { matching: 'material*', target: './vendor.material.$hash.js' },
    ],
    vendor: './vendor.$hash.js',
  },
});
```

My imports are as follow:

packages/app/demo/index.ts
```typescript
  (() => import('@webcomponents/webcomponentsjs/webcomponents-loader'))();
```

packages/app/src/components/button.ts
```typescript
(() => import('@material/mwc-button/mwc-button'))();
```

With this configuration, when building, I'm expecting an output such as:
- app.$hash.js
- vendor.$hash.js
- vendor.material.$hash.js
- vendor.webcomponents.$hash.js
- dynamic/app.$hash.js
- dynamic/button.$hash.js

But unfortunatelly, neither `vendor.material.$hash.js` nor `vendor.webcomponents.$hash.js` are in there. Those dependencies are included in `app.$hash.js` and `dynamic/button.$hash.js`, which are the importers of those dependencies...

I've been debugging the bundler and code-splitting fuse code but couldn't find any reference on where those mappings are being used.

Any clue what I'm doing wrong? I'm quite desperate since I want to improve my app's first load performance and with this bundling configuration it should.

Thanks a lot for your work!!!

**Comment by vegarringdal** [comment]
never used mapping part.
but when I use import() its been auto splitting code into bundles. Auto loaded when I call the function calling import()
been working great for my router.

**Comment by avizcaino** [comment]
The auto splitting only works for me for local components such as the button one in my example, providing a `dynamic/button.$hash.js`. But no for external dependencies imports...

**Comment by avizcaino** [comment]
I've found the section where the bundles are created, and I cannot find any mapping reference in there: https://github.com/fuse-box/fuse-box/blob/f5e9c0967713c78a194fd98a50bbf2bc6d50a5f0/src/bundle/bundleRouter.ts#L96-L118

Making some changes I can generate the expected bundles, but I'm having some problems with modules not found or not referenced properly.

**Comment by nchanged** [comment]
@avizcaino I don't think it has been implemented. I've prepared the interfaces though ;-)
If you are interested, please contribute ;-)

**Comment by avizcaino** [comment]
Hi! @nchanged I've been working on it already in [this PR](https://github.com/fuse-box/fuse-box/pull/1958). Love any feedback you could give me since I'd love this feature to be included in fuse :metal:

**Comment by avizcaino** [comment]
@nchanged so this can be closed right? PR #1958

**Comment by nchanged** [comment]
Awesome! I would like to fix the tests before the release though.
